### DISCUSSION TITLE
[5.x] Add usage information to fields and entry type index/list pages

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#14984] [Ideas] [answered]
It'd be brilliant if the fields and entry type index/list pages (i.e. `/admin/settings/entry-types` and `/admin/settings/fields` included usage information in their tables (similar to the "Used by" panel in field and entry type edit pages).

This information would be especially useful for cleaning up (i.e. deleting) unused fields and types. If full usage info for these pages aren't feasible, even having a binary "In use"/"No usages" column would be very helpful.

**Comment by brandonkelly** [comment]
I originally did this for the Entry Types page in the Craft 5 alpha, but it gets pretty gross for any entry types that have more than a couple usages.

Maybe there’s a middle ground where we only actually display the usage if there’s only one; otherwise show “X usages”.

**Reply by mmikkel** [reply]
> Maybe there’s a middle ground where we only actually display the usage if there’s only one; otherwise show “X usages”.

Being able to _quickly gauge_ if a field or an entry type is _in use or not_, is by far the most important thing IMO. 

If an entry type or field shows "X usages", it's a small feat to click into its edit page to see the list of actual usages there. On the other hand, if it says "No usages", the developer knows it's safe to delete that type or field right from the index page.

...alternatively, maybe multiple usages could be shown inline in the table in a similar manner to how multiple relations are shown inside element indexes:

![CleanShot 2024-05-22 at 00 32 29@2x](https://github.com/craftcms/cms/assets/298510/86987cd4-45f6-4600-b4c1-f1718e21ae68)

**Reply by darwinsoftware** [reply]
I agree with @mmikkel here and want to know the context in which my fields are being used at a glance when managing all fields. The screenshot provided by @mmikkel is exactly what I had in mind with the +X option we see elsewhere in the CMS UI. 

This is an important missing feature for data schema management and identifying deprecated fields. The new abstractions are fantastic and not having field groups makes sense, but it does mean we need to have usage displayed prominently.

**Reply by brandonkelly** [reply]
“+ X” button is a good idea. Just added this for 5.4 (d7fa3dd1e8dbe2ac7061bf964959986b3000b5fc).

**Comment by darwinsoftware** [comment]
Thank you 🙏

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.4.0 is out with new “Used by” columns on the Entry Types and Fields index pages.
